tests/test_classical_mis.py

```diff
@@ -1,13 +1,82 @@
 from typing import Callable
 import networkx as nx
 import pytest
+from pathlib import Path
 
 # Define classical solver configuration
 from mis.solver.solver import MISInstance, MISSolver
 from mis.pipeline.config import SolverConfig
 from mis.shared.types import MethodType
 from mis.pipeline.kernelization import Kernelization
 
+TEST_FILES_DIR = Path.cwd() / "tests/test_files"
+
+
+@pytest.mark.flaky(max_runs=5)
+@pytest.mark.parametrize("dimacs_to_nx", [TEST_FILES_DIR / "a265032_1tc.32.txt"], indirect=True)
+@pytest.mark.parametrize("preprocessor", [None, lambda graph: Kernelization(graph)])
+def test_for_dimacs_32_node_graph(
+    dimacs_to_nx: nx.Graph, preprocessor: None | Callable[[nx.Graph], Kernelization]
+) -> None:
+    """
+    Classical MIS solver for a standard graph benchmark dataset in DIMACS format.
+
+    Can be found here: https://oeis.org/A265032/a265032.html
+    """
+    graph = dimacs_to_nx
+    assert graph.number_of_nodes() == 32
+    assert graph.number_of_edges() == 68
+
+    config = SolverConfig(method=MethodType.EAGER, max_iterations=1, preprocessor=None)
+
+    # Create the MIS instance
+    instance = MISInstance(graph)
+
+    # Run the solver
+    solver = MISSolver(instance, config)
+    solutions = solver.solve().result()
+
+    assert len(solutions[0].nodes) == 12
+
+    # Check the solution is genuinely an independent set.
+    kernel = Kernelization(graph=graph)
+    assert kernel.is_independent(solutions[0].nodes)
+
+
+@pytest.mark.flaky(max_runs=5)
+@pytest.mark.parametrize("dimacs_to_nx", [TEST_FILES_DIR / "a265032_1dc.64.txt"], indirect=True)
+@pytest.mark.parametrize("preprocessor", [None, lambda graph: Kernelization(graph)])
+def test_for_dimacs_64_node_graph(
+    dimacs_to_nx: nx.Graph, preprocessor: None | Callable[[nx.Graph], Kernelization]
+) -> None:
+    """
+    Classical MIS solver for a standard graph benchmark dataset in DIMACS format.
+
+    Can be found here: https://oeis.org/A265032/a265032.html
+    """
+    graph = dimacs_to_nx
+    assert graph.number_of_nodes() == 64
+    assert graph.number_of_edges() == 543
+
+    config = SolverConfig(method=MethodType.EAGER, max_iterations=1, preprocessor=preprocessor)
+
+    # Create the MIS instance
+    instance = MISInstance(graph)
+
+    # Run the solver
+    solver = MISSolver(instance, config)
+    solutions = solver.solve().result()
+
+    # Check the solution is genuinely an independent set.
+    kernel = Kernelization(graph=graph)
+
+    if preprocessor is None:
```

Comment:
You'll need some comment explaining what's going on.

Also, could you check sets of nodes, rather than lists of nodes? This is less likely to break as we add post-processing and/or refactor the solvers.